apis/cache/v1alpha3/redis_types.go

```diff
@@ -56,101 +29,138 @@ const (
 	SupportedRedisVersion = "3.2"
 )
 
-// RedisParameters define the desired state of an Azure Redis cluster. Most
-// fields map directly to an Azure Redis resource:
+// An SKU represents the performance and cost oriented properties of a
+// Redis.
+type SKU struct {
+	//TODO: all three of them required? they might set defaults when sent as empty
+
+	// Name specifies what type of Redis cache to deploy. Valid values: (Basic,
+	// Standard, Premium). Possible values include: 'Basic', 'Standard',
+	// 'Premium'
+	// +kubebuilder:validation:Enum=Basic;Standard;Premium
+	Name string `json:"name"`
+
+	// Family specifies which family to use. Valid values: (C, P). Possible
+	// values include: 'C', 'P'
+	// +kubebuilder:validation:Enum=C;P
+	Family string `json:"family"`
+
+	// Capacity specifies the size of Redis cache to deploy. Valid values: for C
+	// family (0, 1, 2, 3, 4, 5, 6), for P family (1, 2, 3, 4).
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=6
+	Capacity int `json:"capacity"`
+}
+
+// TODO: trial-error the optional fields.
+
+// RedisParameters define the desired state of an Azure Redis cluster.
 // https://docs.microsoft.com/en-us/rest/api/redis/redis/create#redisresource
 type RedisParameters struct {
+	// TODO(muvaf): not included in received object.
 	// ResourceGroupName in which to create this resource.
+	// +immutable
 	ResourceGroupName string `json:"resourceGroupName"`
```

Comment:
Should we introduce a `ResourceGroupNameRef` variant of this field that is an `AttributeReferencer`?